### PR TITLE
 Use XDG_DATA_DIRS to search for soundfonts

### PR DESCRIPTION
**Pull request by gasinvein**
Follow-up for discussion from https://github.com/dreamer/steam-dos/pull/18#discussion_r303206715.

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -208,31 +209,20 @@ def get_dosbox_scaler(self):
...
-            default1_path = os.path.join(*(path + [DEFAULT_MIDI_SOUNDFONT]))
-            default2_path = os.path.join(*(path + ['default.sf2']))
-            if os.path.isfile(selected_path):
-                selected = selected_path
-            if os.path.isfile(default1_path):
-                default = default1_path
-            if os.path.isfile(default2_path):
-                os_default = default2_path
-        use_sf2 = selected or default or os_default
+        data_dirs = os.getenv('XDG_DATA_DIRS', '/usr/share').split(os.pathsep)
```
Comment (dreamer) [inline settings.py]: Put `data_dirs` as a constant in `xdg` module, please.

File: settings.py [hunk 2]
```diff
@@ -208,31 +209,20 @@ def get_dosbox_scaler(self):
...
+        if xdg.DATA_HOME not in data_dirs:
+            data_dirs.insert(0, xdg.DATA_HOME)
+        data_dirs.insert(0, os.path.join(self.distdir, 'share'))
+        use_sf2 = ''
+        sf2_paths = (os.path.join(d, s, n) for n, d, s in itertools.product(
```
Comment (dreamer) [inline settings.py]: This is so nice :)

Comment (dreamer) [reply settings.py]: GitHub selected weird lines when I wanted specifically 217 - I was talking about using `itertools.product` here.

**Comment by gasinvein** [comment]
Sorry for force-pushing, for some reason my local pycodestyle bahave differently from your CI, so I wanted to avoid purely-cosmetic commits.

File: settings.py (hunk 1 above)
Comment (gasinvein) [reply settings.py]: Should it be a string (as-is), or split it into list in `xdg` module?

**Comment by dreamer** [comment]
No worries, I can handle force pushes to private branch :)

CI is set up to use the latest versions of `pylint`, `yapf`, which might be newer than the ones provided by distribution; you can use `pipenv` to setup local virtual environment to use the newest versions:

    $ pipenv install
    $ pipenv shell
    $ make lint
    $ make test
    $ make check-formatting

(I use pipenv only when CI indicates that something is wrong)

File: settings.py (hunk 1 above)
Comment (dreamer) [reply settings.py]: I think splitting it in `xdg` module will be nicer solution.

Comment (gasinvein) [reply settings.py]: Done.

**Comment by dreamer** [comment]
Great change, thanks! :)

**Pull request merged**